docs/dev/stream/testing.md

```diff
@@ -332,6 +332,87 @@ Many more examples for the usage of these test harnesses can be found in the Fli
 
 <span class="label label-info">Note</span> Be aware that `AbstractStreamOperatorTestHarness` and its derived classes are currently not part of the public API and can be subject to change.
 
+### Unit Testing ProcessFunction
+
+The `ProcessFunction` is a widely used low-level and powerful stream processing operation, giving access to the basic building blocks of all (acyclic) streaming applications.
+Flink provides a test harness named `ProcessFunctionTestHarnesses` that can be used to test your `ProcessFunction`. Considering this example:
```

Comment:
I would change this phrase to something like: "Given its importance, in addition to the previous test harnesses that can used directly to test a `ProcessFunction`, Flink provides a test harness factory named `ProcessFunctionTestHarnesses` that allows for easier test harness instantiation. Considering this example:"

Or something along these lines.

In addition, I think it should be mentioned explicitly that in order to use these methods, the user should include the previous changes in his/her `pom` file.